src/services/user/useUser.ts

```diff
@@ -0,0 +1,166 @@
+import { useQuery } from '@tanstack/react-query'
+import { useParams } from 'react-router-dom'
+import { z } from 'zod'
+
+import Api from 'shared/api'
+import { NetworkErrorObject } from 'shared/api/helpers'
+
+const TypeProjectsSchema = z.array(
+  z.union([
+    z.literal('PERSONAL'),
+    z.literal('YOUR_ORG'),
+    z.literal('OPEN_SOURCE'),
+    z.literal('EDUCATIONAL'),
+  ])
+)
+
+const GoalsSchema = z.array(
+  z.union([
+    z.literal('STARTING_WITH_TESTS'),
+    z.literal('IMPROVE_COVERAGE'),
+    z.literal('MAINTAIN_COVERAGE'),
+    z.literal('TEAM_REQUIREMENTS'),
+    z.literal('OTHER'),
+  ])
+)
+
+const MeSchema = z.object({
+  owner: z.object({
+    defaultOrgUsername: z.string().nullable(),
+  }),
+  email: z.string().nullable(),
+  privateAccess: z.boolean().nullable(),
+  onboardingCompleted: z.boolean(),
+  businessEmail: z.string().nullable(),
+  termsAgreement: z.boolean().nullable(),
+  user: z.object({
+    name: z.string().nullable(),
+    username: z.string(),
+    avatarUrl: z.string(),
+    avatar: z.string(),
+    student: z.boolean(),
+    studentCreatedAt: z.string().nullable(),
+    studentUpdatedAt: z.string().nullable(),
+    customerIntent: z.string().nullable(),
+  }),
+  trackingMetadata: z.object({
+    service: z.string(),
+    ownerid: z.number(),
+    serviceId: z.string(),
+    plan: z.string().nullable(),
+    staff: z.boolean().nullable(),
+    hasYaml: z.boolean(),
+    bot: z.string().nullable(),
+    delinquent: z.boolean().nullable(),
+    didTrial: z.boolean().nullable(),
+    planProvider: z.string().nullable(),
+    planUserCount: z.number().nullable(),
+    createdAt: z.string().nullable(),
+    updatedAt: z.string().nullable(),
+    profile: z
+      .object({
+        createdAt: z.string(),
+        otherGoal: z.string().nullable(),
+        typeProjects: TypeProjectsSchema,
+        goals: GoalsSchema,
+      })
+      .nullable(),
+  }),
+})
+
+type Me = z.infer<typeof MeSchema>
+
+const UserSchema = z.object({
+  me: MeSchema.nullable(),
+})
+
+export type User = z.infer<typeof UserSchema>
+
+const currentUserFragment = `
+fragment CurrentUserFragment on Me {
+  owner {
+    defaultOrgUsername
+  }
+  email
+  privateAccess
+  onboardingCompleted
+  businessEmail
+  termsAgreement
+  user {
+    name
+    username
+    avatarUrl
+    avatar: avatarUrl
```

Comment:
Are these the same? Do we need both?

Reply:
They are the same. We had both in the old query and so I decided to just keep them. It seems like we use both. I will make a ticket to go back and consolidate this though as it's definitely no ideal lol